met/src/tools/other/ioda2nc/ioda2nc.cc

```diff
@@ -775,7 +776,14 @@ void process_ioda_file(int i_pb) {
 
       n_hdr_obs = 0;
       for(idx=0; idx<v_obs_data.size(); idx++ ) {
-         obs_arr[1] = idx;
+         int var_idx;
+         if (!obs_var_names.has(raw_var_names[idx], var_idx)) {
+            mlog << Warning << "\n" << method_name
+                 << "Skip the variable " << raw_var_names[idx]
+                 << " at " << ioda_files[i_pb] << "\n\n";
+            continue;
+         }
+         obs_arr[1] = var_idx;
```

Comment:
Is var_idx uninitialized at this point?

Reply:
It's set by the next line: "if (!obs_var_names.has(raw_var_names[idx], var_idx))" but it 's safe to initialized before updating it. It's corrected.

Reply:
I looked into the StringArray::has function and it looks like this logic sets var_idx to the index where the value was found. Sorry, I misunderstood how the function worked.